NitroxServer-Subnautica/GameLogic/Entities/Spawning/SubnauticaEntitySpawnPointFactory.cs

```diff
@@ -8,36 +9,64 @@ namespace NitroxServer_Subnautica.GameLogic.Entities.Spawning
 {
     public class SubnauticaEntitySpawnPointFactory : EntitySpawnPointFactory
     {
+        private readonly Dictionary<string, EntitySpawnPoint> spawnPointsByUid = new Dictionary<string, EntitySpawnPoint>();
+
         public override List<EntitySpawnPoint> From(AbsoluteEntityCell absoluteEntityCell, Transform transform, GameObject gameObject)
         {
             List<EntitySpawnPoint> spawnPoints = new List<EntitySpawnPoint>();
             EntitySlotsPlaceholder entitySlotsPlaceholder = gameObject.GetComponent<EntitySlotsPlaceholder>();
+            
 
             if (!ReferenceEquals(entitySlotsPlaceholder, null))
             {
                 foreach (EntitySlotData entitySlotData in entitySlotsPlaceholder.slotsData)
                 {
                     List<EntitySlot.Type> slotTypes = SlotsHelper.GetEntitySlotTypes(entitySlotData);
                     List<string> stringSlotTypes = slotTypes.Select(s => s.ToString()).ToList();
-
-                    spawnPoints.Add(
-                        new EntitySpawnPoint(absoluteEntityCell,
+                    EntitySpawnPoint entitySpawnPoint = new EntitySpawnPoint(absoluteEntityCell,
                                              entitySlotData.localPosition,
                                              entitySlotData.localRotation,
                                              stringSlotTypes,
                                              entitySlotData.density,
-                                             entitySlotData.biomeType.ToString())
-                    );
+                                             entitySlotData.biomeType.ToString());
+
+
+                    BuildChildren(entitySpawnPoint, gameObject);
+                    if (gameObject.Parent == null)
+                    {
+                        spawnPoints.Add(
+                            entitySpawnPoint
+                        );
+                    }
                 }
             }
             else
             {
-                spawnPoints.Add(
-                    new EntitySpawnPoint(absoluteEntityCell, transform.Position, transform.Rotation, transform.Scale, gameObject.ClassId)
-                );
+                EntitySpawnPoint entitySpawnPoint = new EntitySpawnPoint(absoluteEntityCell, transform.Position, transform.Rotation, transform.Scale, gameObject.ClassId);
+
+                BuildChildren(entitySpawnPoint, gameObject);
+
+                if (gameObject.Parent == null)
+                {
+                    spawnPoints.Add(
+                        entitySpawnPoint
+                    );
+                }
             }
 
             return spawnPoints;
         }
+
+        private void BuildChildren(EntitySpawnPoint entitySpawnPoint, GameObject gameObject)
```

Comment:
Might be more explicit as `AssignParent` or maybe `HandleParenting` (and in that case, you can move the null check/spawnPoint addition code into here)

Reply:
Renamed to HandleParenting and moved null check